File: src/content/docs/topics/cs/coding-problems/stack/394-decode-string.go
package main

import "fmt"

func assert(condition bool, msgs ...string) {
	if !condition {
		msg := "assertion failed"
		if len(msgs) > 0 {
			msg = msgs[0]
		}
		panic(msg)
	}
}

func decodeString(s string) string {
	// TODO: implement
	return ""
}

func runTests() {
	assert(decodeString("3[a]2[bc]") == "aaabcbc")
	assert(decodeString("3[a2[c]]") == "accaccacc")
	assert(decodeString("2[abc]3[cd]ef") == "abcabccdcdcdef")
	fmt.Println("all tests pass")
}

func main() { runTests() }

File: src/content/docs/topics/cs/coding-problems/stack/394-decode-string.ts
function assert(condition: boolean, msg: string = ''): void {
    if (!condition) throw new Error(msg || 'Assertion failed');
}

function decodeString(s: string): string {
    // TODO: implement
    return '';
}

function _runTests(): void {
    assert(decodeString('3[a]2[bc]') === 'aaabcbc');
    assert(decodeString('3[a2[c]]') === 'accaccacc');
    assert(decodeString('2[abc]3[cd]ef') === 'abcabccdcdcdef');
    assert(decodeString('a') === 'a');
    console.log('all tests pass');
}

_runTests();
